synapse/storage/databases/main/events_worker.py

```diff
@@ -754,7 +754,31 @@ def _invalidate_local_get_event_cache(self, event_id: str) -> None:
     async def _get_events_from_cache(
         self, events: Iterable[str], update_metrics: bool = True
     ) -> Dict[str, EventCacheEntry]:
-        """Fetch events from the caches.
+        """Fetch events from the caches, both in memory and any external.
+
+        May return rejected events.
+
+        Args:
+            events: list of event_ids to fetch
+            update_metrics: Whether to update the cache hit ratio metrics
+        """
+        event_map = self._get_events_from_local_cache(events, update_metrics=update_metrics)
```

Comment:
it seems like we're going to end up updating the metrics on `_get_event_cache` twice here (though I'm rather wondering if a single set of metrics is actually meaningful for this two-level cache)